Assets/Scripts/EnemyAI.cs

```diff
@@ -15,16 +14,19 @@ public class EnemyAI : MonoBehaviour
     [SerializeField] float detectRange = 5;
     [SerializeField] float hearRange = 1.5f;
     float _attackRange;
-    [SerializeField] GameObject bullet;
-    Bullet _bullet;
-    [SerializeField] float shootDelay = 1;
-    bool _canShoot = true;
+    [SerializeField] GameObject projectile;
+    [SerializeField] float attackDelay = 1;
+    bool _canAttack = true;
+    Health _playerHealth;
+    LayerMask _playerLayer;
 
     void Awake()
     {
+        _player = GameObject.FindWithTag("Player").transform;
         _speed = patrolSpeed;
-        _attackRange = type == Type.Melee ? 0.5f : 3.5f;
-        _bullet = bullet.GetComponent<Bullet>();
+        _attackRange = type == Type.Melee ? 1.5f : 3.5f;
+        _playerLayer = LayerMask.GetMask("Player");
```

Comment:
currently attack range is hardcoded, moreover it might depend not only on type but also on different concrete enemies: guard, shooter, watcher and more in the future. We also will require custom behaviors.

The way I imagined it is way more generic. Like having 
abstract class EnemyAttact , 
abstract class EnemyRangedAttack : EnemyAttact 
abstract class EnemyMeleeAttack : EnemyAttact 
class DarknightAttack : EnemyMeleeAttack 
etc.

but that's something we can consider in the future. Feel free to merge your PR. You did a great job!!